source/common/common/utility.cc

```diff
@@ -102,6 +102,30 @@ std::string StringUtil::subspan(const std::string& source, size_t start, size_t
   return source.substr(start, end - start);
 }
 
+std::vector<std::string> StringUtil::tokenize(const std::string& source,
```

Comment:
This is a more generalized version of StringUtil::split(). I'm not sure if we really need both? Should we get rid of split() and just call tokenize split and make it take multiple splitters? This one will be a bit slower, but not sure if it's worth having both versions.

Either way, you need test cases for this function in utility_test.cc.